### PR TITLE
Improve property escaping

### LINKED ISSUE #1560: Code generation fails with the following error
I am getting compilation errors when generating the source code. 

Error:
 public final BooleanPath 8FRecord = createBoolean("8FRecord");

COMPILATION ERROR :Return type for the method is missing
	- Syntax error, insert ")" to complete ConstructorDeclaration
	- Syntax error on tokens, AnnotationName expected instead
	- Syntax error, insert "Identifier (" to complete 
	 ConstructorHeaderName

Can some one please help me to fix that.

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1560

### CODE REVIEW

File: querydsl-codegen/src/main/java/com/querydsl/codegen/Property.java [hunk 1]
```diff
@@ -65,6 +64,16 @@ public Property(EntityType declaringType, String name, String escapedName, Type
         this.inherited = inherited;
     }
 
+    private static String escapeName(String name) {
+        if (JavaSyntaxUtils.isReserved(name)) {
+            name = name + "$";
+        }
+        if (!Character.isJavaIdentifierStart(name.charAt(0))) {
```
Comment (Shredder121) [inline querydsl-codegen/src/main/java/com/querydsl/codegen/Property.java]: They are mutually exclusive?
Or is that a micro optimization?

Comment (timowest) [inline querydsl-codegen/src/main/java/com/querydsl/codegen/Property.java]: Good point.

**Comment by Shredder121** [comment]
To me this looks good.

**Comment by johnktims** [comment]
This looks good to me, too.

**Pull request merged**